tests/PHPStan/Rules/Methods/CallStaticMethodsRuleTest.php

```diff
@@ -432,17 +432,36 @@ public function testBug1971(): void
 	{
 		$this->checkThisOnly = false;
 		$this->analyse([__DIR__ . '/data/bug-1971.php'], [
+			[
+				'Parameter #1 $callback of static method Closure::fromCallable() expects callable(): mixed, array{\'Bug1971\\\HelloWorld\', \'sayHello\'} given.',
+				14,
+			],
+			[
+				'Parameter #1 $callback of static method Closure::fromCallable() expects callable(): mixed, array{class-string<static(Bug1971\\HelloWorld)>, \'sayHello\'} given.',
+				15,
+			],
```

Comment:
Did not update the test data, but cover the fact these instance methods called statically do not exist.

Reply:
The problem here is that this actually works: https://3v4l.org/mTa9S

It only doesn't work from the outside: https://3v4l.org/dUmHk

The same problem is covered in CallStaticMethodRule here: https://github.com/phpstan/phpstan-src/blob/625cb846a14e7a401534c04442e1f08cfd90251b/src/Rules/Methods/CallStaticMethodsRule.php#L215-L236

So we should replicate the same logic here. We can take advantage of Scope being passed to `getCallableParametersAcceptors` and also pass it along to `findTypeAndMethodName` but as an optional parameter because of backward compatibility (but go through all the current call sites and pass Scope in it).

Reply:
One problem I'm hitting: `isCallable` does not have a scope, and that's where `findTypeAndMethodName` seems to be invoked from. So I added `OutOfClassScope`, to check if we're in a class. If not, return `null`, but that always returns false.

Reply:
Having ConstantArrayType return `maybe` on a static call outside of the class or in class with mismatching class names seems to have resolved this. Thanks for the tip!